dev/com.ibm.ws.concurrent.persistent_fat_multiple/fat/src/com/ibm/ws/concurrent/persistent/fat/multiple/DatabaseContainerFactory.java

```diff
@@ -0,0 +1,126 @@
+package com.ibm.ws.concurrent.persistent.fat.multiple;
+
+import java.time.Duration;
+
+import org.testcontainers.containers.Db2Container;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.MSSQLServerContainer;
+import org.testcontainers.containers.OracleContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.output.OutputFrame;
+
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.custom.junit.runner.FATRunner;
+
+/**
+ * This is a factory class that creates a database test container.
+ * The test container returned will be based on the {fat.bucket.db.type} 
+ * system property. 
+ * 
+ * The {fat.bucket.db.type} property is set to different databases
+ * by our test infrastructure when a fat-suite has the property
+ * {fat.test.databases} set to true.
+ * 
+ * <br> Container Information: <br>
+ * DERBY: Uses a derby no-op test container <br>
+ * DB2: Uses <a href="https://hub.docker.com/r/ibmcom/db2">Offical DB2 Container</a> <br>
+ * Oracle: TODO replace this container with the official oracle-xe container if/when it is available without a license. <br>
+ * Postgre: Uses <a href="https://hub.docker.com/_/postgres">Offical Postgres Container</a> <br>
+ * MS SQL Server: Uses <a href="https://hub.docker.com/_/microsoft-mssql-server">Offical Microsoft SQL Container</a> <br>
+ * 
+ * TODO: Since this factory class is general it should be moved to fattest simplicity in the future
+ */
+public class DatabaseContainerFactory {	
+	private static final Class<DatabaseContainerFactory> c = DatabaseContainerFactory.class;
+	private static JdbcDatabaseContainer<?> container;
+	private static DatabaseContainerType dbContainerType;
+	
+	/**
+	 * @see #create(String databaseName)
+	 * 
+	 * Uses a default database name 'test'
+	 */
+	public static JdbcDatabaseContainer<?> create() {
+		return DatabaseContainerFactory.create("test");
+	}
+	
+	/**
+	 * Reads the {fat.bucket.db.type} system property and returns a
+	 * container based on that property. 
+	 * [Postgre, DB2, Oracle, SQLServer, Derby Embedded]
+	 * 
+	 * If {fat.bucket.db.type} is not set with a value, 
+	 * we will default to Derby Embedded.
+	 * 
+	 * @param databaseName - String passed to database container for those database containers 
+	 * that can use a custom database.
+	 *  
+	 * @return JdbcDatabaseContainer - The test container.
+	 */
+	public static JdbcDatabaseContainer<?> create(String databaseName) {
```

Comment:
I just realized that nothing is requiring `fat.test.databases: true` to be set when running locally, but this property needs to be set in order for the bucket to be picked up by the DB rotation build. So lets take the opportunity at the beginning of this method to make sure `fat.test.databases: true` is set and blow up with a message like:
`To use a generic database, the FAT must be opted into DB rotation by setting 'fat.test.databases: true' in the FAT project's bnd.bnd file`
if it's not set. This should guard against people getting everything working locally and then wondering why the bucket isn't running in DB rotation (or just not even realizing it's not running in rotation)